docs/compose-file.md

```diff
@@ -41,9 +55,47 @@ Compose will build and tag it with a generated name, and use that image thereaft
 
     build: /path/to/build/dir
 
-Using `build` together with `image` is not allowed. Attempting to do so results in
+    build:
+      context: /path/to/build/dir
+
+Using `context` together with `image` is not allowed. Attempting to do so results in
 an error.
 
+#### dockerfile
+
+Alternate Dockerfile.
+
+Compose will use an alternate file to build with. A build path must also be
+specified using the `build` key.
+
+    build:
+      context: /path/to/build/dir
+      dockerfile: Dockerfile-alternate
+
+Using `dockerfile` together with `image` is not allowed. Attempting to do so results in an error.
+
+
+#### args
+
+Add build arguments. You can use either an array or a dictionary. Any
+boolean values; true, false, yes, no, need to be enclosed in quotes to ensure
+they are not converted to True or False by the YML parser.
+
+Build arguments with only a key are resolved to their environment value on the
+machine Compose is running on, which can be helpful for secret or host-specific values.
```

Comment:
The [dockerfile docs](https://docs.docker.com/engine/reference/builder/#arg) say not to use this for secrets. The reasons aren't really clear (maybe the values are stored in the image after?), but I think it's best we don't contradict the documentation.